### PR TITLE
PP-10639: Remove status from agreement events

### PR DESCRIPTION
**Pull request by alan-gds**
Ledger is no longer going to use this field, so it is now redundant; instead, Ledger itself will define the mapping from event types to `AgreementTypes`.

~https://github.com/alphagov/pay-ledger/pull/2425 needs to be deployed to production before this PR can be merged.~ done

### CODE REVIEW

**Review by sfount — COMMENTED** [review]
Change looks great, really nice cleaning up with the events package move 👍 

This shouldn't be merged until Ledger is ready to map these events here https://github.com/alphagov/pay-ledger/pull/2425

File: src/main/java/uk/gov/pay/connector/events/model/agreement/AgreementCancelledByService.java [hunk 1]
```diff
@@ -17,20 +17,12 @@ public static AgreementCancelledByService from(AgreementEntity agreement, Instan
                 agreement.getServiceId(),
                 agreement.getGatewayAccount().isLive(),
                 agreement.getExternalId(),
-                new AgreementCancelledByServiceEventDetails(AgreementStatus.CANCELLED),
+                new AgreementCancelledByServiceEventDetails(),
                 timestamp
         );
     }
 
     static class AgreementCancelledByServiceEventDetails extends EventDetails {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/events/model/agreement/AgreementCancelledByService.java]: I don't think we need this nested class if there are no event details. There is an overridden constructer in `AgreementEvent` that doesn't take an EventDetails

Comment (alan-gds) [reply src/main/java/uk/gov/pay/connector/events/model/agreement/AgreementCancelledByService.java]: Fair enough; I will change this.

**Review by stephencdaly — APPROVED** [review]
LGTM!

**Pull request merged**